snuba/query/data_source/simple.py

```diff
@@ -45,7 +45,7 @@ class Table(SimpleDataSource):
 
     table_name: str
     schema: ColumnSet
-    final: bool = False
+    final: bool = field(default=False, compare=False)
```

Comment:
Why did you need this?

Reply:
When running tests, there was this single test which failed. https://github.com/getsentry/snuba/blob/master/tests/pipeline/test_composite_planner.py#L466
The difference was the final member field. 

I think the final field should not matter in equality tests since its more a matter of applying optimization. Let me know if you think we should to consider it for equality tests and I can then fix the tests

Reply:
How about updating the processed query in the test_composite_planner to just have final=True like the processing pipeline will produce so this isn't required. https://github.com/getsentry/snuba/blob/master/tests/pipeline/test_composite_planner.py#L71

If we can avoid it I think we should avoid skipping the comparison on this field, as it causes two queries that are not the same to be (arguably incorrectly) reported as equal. I think it will make it harder to write tests that do check for the final behavior, which we may want to have. This code is not test specific so I also worry that it introduces the potential for incorrect comparisons at runtime.